File: src/main/scala/net/machinemuse/numina/render/MuseTESR.scala
package net.machinemuse.numina.render

import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.ResourceLocation

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 4:28 PM, 8/3/13
 */
abstract class MuseTESR[T <: TileEntity] extends TileEntitySpecialRenderer{
  def bindTextureByName(tex: String) {
    this.bindTexture(new ResourceLocation(tex))
  }

  def renderTileEntityAt(tileentity: TileEntity, d0: Double, d1: Double, d2: Double, f: Float): Unit = {
    renderAt(tileentity, d0, d1, d2, f)
  }
  def renderAt(tileentity: TileEntity, d0: Double, d1: Double, d2: Double, f: Float):Unit
}

File: src/main/java/net/machinemuse/numina/render/MuseTESR.java
package net.machinemuse.numina.render;

import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.ResourceLocation;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 4:28 PM, 8/3/13
 *
 * Ported to Java by lehjr on 10/23/16.
 */
public class MuseTESR extends TileEntitySpecialRenderer {
    public void bindTextureByName(String tex) {
        this.bindTexture(new ResourceLocation(tex));
    }

    @Override
    public void renderTileEntityAt(TileEntity tileentity, double d0, double d1, double d2, float f) {
        this.renderAt(tileentity, d0, d1, d2, f);
    }

    public void renderAt(TileEntity var1, double var2, double var4, double var6, float var8) {

    }
}
